src/test/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/connection/TestConnectionManager.kt

```diff
@@ -1190,6 +1192,82 @@ class TestConnectionManager {
         )
     )
 
+    @Test
+    fun `connection manager receives messages when publishing is blocked`() {
+        val routingKey = "routingKey1"
+        val queueName = "queue1"
+        val exchange = "test-exchange1"
+        rabbit.let { rabbit ->
+            declareQueue(rabbit, queueName)
+            declareFanoutExchangeWithBinding(rabbit, exchange, queueName)
+
+            LOGGER.info { "Started with port ${rabbit.amqpPort}" }
+            LOGGER.info { "Started with port ${rabbit.amqpPort}" }
+            val messagesCount = 10
+            val blockAfter = 3
+            val countDown = CountDownLatch(messagesCount)
+            val messageSizeBytes = 7
+            createConnectionManager(
+                rabbit, ConnectionManagerConfiguration(
+                    subscriberName = "test",
+                    prefetchCount = DEFAULT_PREFETCH_COUNT,
+                    confirmationTimeout = DEFAULT_CONFIRMATION_TIMEOUT,
+                    enablePublisherConfirmation = true,
+                    maxInflightPublicationsBytes = messagesCount * messageSizeBytes,
+                    heartbeatIntervalSeconds = 1,
+                    minConnectionRecoveryTimeout = 2000,
+                    maxConnectionRecoveryTimeout = 2000,
+                    // to avoid unexpected delays before recovery
+                    retryTimeDeviationPercent = 0
+                )
+            ).use { manager ->
+                repeat(messagesCount) { index ->
```

Comment:
messagesCount is greater than blockAfter, it means that publisher not blocking in this test because check logic is after this circle

Reply:
We send 3 messages, then block publishers, then send 7 more. Subscribe, receive 3, unblock, receive 7. This allows to ensure that publishers were actually blocked. We can simplify the scenario: just send a few messages, block publishers, try to read messages.

Reply:
`then send 7 more` -> If publisher blocked we could able to publish additional messages